src/continuous_agg.c

```diff
@@ -233,6 +264,68 @@ ts_materialization_invalidation_log_delete(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+static void
+continuous_agg_fill_bucket_function(int32 mat_hypertable_id, ContinuousAggsBucketFunction *bf)
+{
+	ScanIterator iterator;
+	int count = 0;
+
+	iterator = ts_scan_iterator_create(CONTINUOUS_AGGS_BUCKET_FUNCTION,
+									   AccessShareLock,
+									   CurrentMemoryContext);
+	init_scan_cagg_bucket_function_by_mat_hypertable_id(&iterator, mat_hypertable_id);
+	ts_scanner_foreach(&iterator)
+	{
+		Datum values[Natts_continuous_aggs_bucket_function];
+		bool isnull[Natts_continuous_aggs_bucket_function];
+
+		bool should_free;
+		HeapTuple tuple = ts_scan_iterator_fetch_heap_tuple(&iterator, false, &should_free);
+
+		/*
+		 * Our usual GETSTRUCT() approach doesn't work when TEXT fields are involved,
+		 * thus a more robust approach with heap_deform_tuple() is used here.
+		 */
+		heap_deform_tuple(tuple, ts_scan_iterator_tupledesc(&iterator), values, isnull);
+
+		Assert(!isnull[Anum_continuous_aggs_bucket_function_experimental - 1]);
+		bf->experimental =
+			DatumGetBool(values[Anum_continuous_aggs_bucket_function_experimental - 1]);
+
+		Assert(!isnull[Anum_continuous_aggs_bucket_function_name - 1]);
```

Comment:
I don't particularly like this because one silent disk corruption will cause the extension to crash. I would prefer an exception to be thrown.

Reply:
I don't think I can agree. In case of disc corruption (and assuming heap checksums are turned off) the crash of TimescaleDB will be the least of the user's problems. Not mentioning that TimescaleDB is not the right place to check for disc corruption. If we start doing this there will be so many if's in the code that we can forget about the performance.

Reply:
Even after the disk has been repaired the TimescaleDB extension will still crash so it doesn't really matter what the right place for corruption handling is. This is not a transient bit-flip that will correct itself after the process restarts. This will keep crashing every time. Obviously, I don't feel strongly about this, but I am interested in how this is handled in our extension in general.

Reply:
> Obviously I don't feel strongly about this, but I am interested in how this is handled in our extension in general.

I have some experience with PostgreSQL extensions and also a very unpleasant experience of restoring databases from corrupted disks... To my knowledge, no one handles this in the extensions. I don't think this is even handled in PostgreSQL core, except for the heap checksums. (The core is huge thus I can't be 100% sure.)

Typically when the disk corruption is detected what you do is remove the DBMS instance from the cluster, repair the hardware, restore the instance from the latest backup and fill it using synchronous replication. If you don't have a backup (sadly this happens too) either your data was not that valuable and you create a new database, or you restore what you can manually, using tools like pg_filedump.